nifi-nar-bundles/nifi-standard-bundle/nifi-standard-processors/src/main/java/org/apache/nifi/processors/standard/CountText.java

```diff
@@ -213,10 +220,10 @@ public void onTrigger(ProcessContext context, ProcessSession processSession) thr
         }
         AtomicBoolean error = new AtomicBoolean();
 
-        lineCount = 0;
-        lineNonEmptyCount = 0;
-        wordCount = 0;
-        characterCount = 0;
+        final int[] lineCount = {0};
+        final int[] lineNonEmptyCount = {0};
+        final int[] wordCount = {0};
+        final int[] characterCount = {0};
```

Comment:
Not necessarily, I think it is better to use an array than an AtomicReference <Object> for local use in a lambda expression, however I can change it to AtomicInteger.

Reply:
what is your opinion?

Reply:
I understand that your reading in the code can be confusing.

Reply:
I personally think that a primitive variable is better than a synchronized one when we talk about its use in a local context that does not require synchronism. But I understand that this can be confusing.

Reply:
just for performance